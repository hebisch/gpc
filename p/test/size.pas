program size(input,output);

type
str = array[1..20] of char;
bigary = array[1..10000] of integer;
{huge = array[1..20000] of integer;}
color = (black, purple, violet, red, pink, green,
         blue, orange, yellow, white, none);
const
colornames:array[color] of string[10] =
('black', 'purple', 'violet', 'red', 'pink', 'green',
         'blue', 'orange', 'yellow', 'white', 'none');

var
nums :bigary;
{mnums :huge;}
paint :color;
s: string (10);

function bar( value :integer) :str;
begin
write(value:1);
bar := ' is the value';
end;

function foo :str;
begin
foo := 'this is a test';
end;

begin
writeln('Enter an integer');
read(nums[5]);
writeln(bar(nums[5]));
writeln(foo);
writeln('enter a color');
while input^=' ' do get(input);
read(s);
paint:=black;
while (paint<none) and (colornames[paint]<>s) do Inc(paint);
writeln('The next color is ',colornames[succ(paint)]);
end.
