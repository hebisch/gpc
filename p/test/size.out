Enter an integer
17 is the value       
this is a test      
enter a color
The next color is yellow
